crates/bevy_render/src/texture/image_texture_conversion.rs

```diff
@@ -1,156 +1,49 @@
-use crate::texture::{Image, TextureFormatPixelInfo};
+use crate::texture::Image;
 use anyhow::anyhow;
-use image::{DynamicImage, ImageBuffer};
+use image::{buffer::ConvertBuffer, DynamicImage, ImageBuffer, Luma, Rgba};
 use wgpu::{Extent3d, TextureDimension, TextureFormat};
 
 impl Image {
     /// Converts a [`DynamicImage`] to an [`Image`].
     pub fn from_dynamic(dyn_img: DynamicImage, is_srgb: bool) -> Image {
-        use bevy_core::cast_slice;
-        let width;
-        let height;
+        let width = dyn_img.width();
+        let height = dyn_img.height();
 
         let data: Vec<u8>;
         let format: TextureFormat;
 
         match dyn_img {
             DynamicImage::ImageLuma8(i) => {
-                let i = DynamicImage::ImageLuma8(i).into_rgba8();
-                width = i.width();
-                height = i.height();
-                format = if is_srgb {
-                    TextureFormat::Rgba8UnormSrgb
-                } else {
-                    TextureFormat::Rgba8Unorm
-                };
-
+                format = TextureFormat::R8Unorm;
                 data = i.into_raw();
             }
             DynamicImage::ImageLumaA8(i) => {
-                let i = DynamicImage::ImageLumaA8(i).into_rgba8();
-                width = i.width();
-                height = i.height();
-                format = if is_srgb {
-                    TextureFormat::Rgba8UnormSrgb
-                } else {
-                    TextureFormat::Rgba8Unorm
-                };
-
-                data = i.into_raw();
+                format = TextureFormat::R8Unorm;
+                data = ConvertBuffer::<ImageBuffer<Luma<u8>, Vec<u8>>>::convert(&i).into_raw();
             }
             DynamicImage::ImageRgb8(i) => {
-                let i = DynamicImage::ImageRgb8(i).into_rgba8();
-                width = i.width();
-                height = i.height();
-                format = if is_srgb {
-                    TextureFormat::Rgba8UnormSrgb
+                if is_srgb {
+                    format = TextureFormat::Rgba8UnormSrgb;
                 } else {
-                    TextureFormat::Rgba8Unorm
-                };
-
-                data = i.into_raw();
+                    format = TextureFormat::Rgba8Unorm;
+                }
+                data = ConvertBuffer::<ImageBuffer<Rgba<u8>, Vec<u8>>>::convert(&i).into_raw();
             }
             DynamicImage::ImageRgba8(i) => {
-                width = i.width();
-                height = i.height();
-                format = if is_srgb {
-                    TextureFormat::Rgba8UnormSrgb
+                if is_srgb {
+                    format = TextureFormat::Rgba8UnormSrgb;
                 } else {
-                    TextureFormat::Rgba8Unorm
-                };
-
-                data = i.into_raw();
-            }
-            DynamicImage::ImageLuma16(i) => {
```

Comment:
Could you explain why you removed support for the following image formats: `Luma16`, `LumaA16`, `Rgb16`, `Rgba16`, and 32F variants?

Reply:
Sorry for not providing enough context and information at the first time.

I'm creating a 3D asset management tool based on bevy, it should be able to load and preview gltf 2.0 model at runtime, so the stability and compatibility comes to first priority.

I've tested the compatibility with all kinds of gltf, including all models from (https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0) and models I collected which will cause the crash issue #7005, and current code will provide maximum stability and compatibility, means all the textures could be loaded and rendered without any problem.

I think, we could build a solid ground without any crash at first, and extend the compatibility case-by-case in future, rather than support as much as possible.

Reply:
What about custom shaders? Bevy users might want to load any image formats for their own shaders. Even if it doesn't work with the default PBR shader